### PR TITLE
Documentation: document how to debug runtime with Delve

### PR DESCRIPTION
**Pull request by aarzilli**
Add FAQ entry to document how to debug the Go runtime using Delve.

Also add table of contents to FAQ.

### CODE REVIEW

File: _scripts/gen-faq-toc.go [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+)
+
+func must(err error, fmtstr string, args ...interface{}) {
+	if err != nil {
+		log.Fatalf(fmtstr, args...)
+	}
+}
+
+func usage() {
+	fmt.Fprintf(os.Stderr, "gen-faq-to input-path output-path")
```
Comment (alexandear) [inline _scripts/gen-faq-toc.go:21]: Why not just use `os.Stderr`?

```suggestion
	os.Stderr.WriteString("gen-faq-to input-path output-path")
```

File: _scripts/gen-faq-toc.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+	anchor, title string
+}
+
+const (
+	startOfMappingTable = "<!-- BEGIN TOC -->"
+	endOfMappingTable   = "<!-- END TOC -->"
+)
+
+func spliceDocs(docpath string, docs []byte, outpath string) {
+	docbuf, err := ioutil.ReadFile(docpath)
```
Comment (alexandear) [inline _scripts/gen-faq-toc.go:37]: `ioutil` is deprecated since [Go 1.16](https://go.dev/doc/go1.16#ioutil). Let's use `os.ReadFile` instead.

Comment (aarzilli) [reply _scripts/gen-faq-toc.go:37]: Not worth compromising compilation on older versions of Go for a deprecation.

**Comment by aarzilli** [comment]
Done.

**Review by derekparker — APPROVED** [review]
LGTM

**Pull request merged**